### PR TITLE
Companies local nav

### PR DESCRIPTION
**Pull request by teneightfive**
Replace companies nav with newer dynamic macro.

This removes the manual markup for the nav and the need to send the tab to each view in order to highlight it.

### CODE REVIEW

**Comment by tyom** [comment]
We've been calling details middleware `details.js` in other places. I'm not against `params.js` but we should make it consistent.

File: src/apps/companies/middleware/params.js [hunk 1]
```diff
@@ -17,9 +18,13 @@ function getHeadingAddress (company) {
   return getFormattedAddress(company, 'registered')
 }
 
-async function getCompany (req, res, next, id) {
+async function getCompany (req, res, next, companyId) {
+  if (!isValidGuid(companyId)) {
```
Comment (tyom) [inline src/apps/companies/middleware/params.js]: Problem is your next controller in the chain expects a company and we get `500`. I think we should let `gitDitCompany` throw and let our error middleware handle `404` which is more appropriate for invalid id.

**Comment by teneightfive** [comment]
>We've been calling details middleware details.js in other places. I'm not against params.js but we should make it consistent.

It's not specific to details though. It's middleware to house param functions. I've used this pattern in OMIS so it's not different to what exists. I feel `details` doesn't really say that much and makes me think its only for the details/overview route, not shared across many routes.

File: src/apps/companies/middleware/params.js (hunk 1 above)
Comment (teneightfive) [reply src/apps/companies/middleware/params.js]: Will remove commit.

**Pull request merged**